segmentstore/contracts/src/main/java/io/pravega/segmentstore/contracts/StreamSegmentStore.java

```diff
@@ -155,11 +155,11 @@
      * @param targetSegmentName The name of the StreamSegment to merge into.
      * @param sourceSegmentName The name of the StreamSegment to merge.
      * @param timeout           Timeout for the operation.
-     * @return A CompletableFuture that, when completed normally, will indicate the operation completed. If the operation
-     * failed, the future will be failed with the causing exception.
+     * @return A CompletableFuture that, when completed normally, will contain a SegmentProperties instance with the last known
```

Comment:
Because PRP needs to log some metrics related to the source segment, such as the number of events merged and size. We need to return this now since after the fact the source segment won't be available. We also can't check it beforehand since there may be uncommitted operations on the source which may affect its properties.